File: ruby/account_details.rb
# This example shows a basic GET request to get all account data
require 'awesome_print'
require './lib/mailchimp'

# Do we have an API Key? If so, initialize a new Mailchimp object
if File.exist?('../APIKEY')
  @api_key = File.open('../APIKEY').read.strip
  @mailchimp_client = Mailchimp.new(@api_key)
else
  raise StandardError, "APIKEY file is not present, please read README.md for more instructions"
end

response = @mailchimp_client.account_details

puts "-----\nHEADERS\n"
ap response.headers
puts "\n\n-----\nACCOUNT DETAILS\n"
ap response

File: php/account-details.php
<?php

    require_once('lib/MailChimp.php');
    $apiKeyPath = '../APIKEY';

    if (file_exists($apiKeyPath)) {
        $apiKey = file_get_contents($apiKeyPath);
        $MailChimp = new MailChimp(YOUR_MAILCHIMP_API_USER, $apiKey);
    } else {
        throw new Exception('Missing an API key in a file called "APIKEY" in the root folder', 1);
    }

    $response = $MailChimp->accountDetails();
    var_dump($response);

?>